Improve maps_generator README. (#4404)
organicmaps/organicmaps: tools/python/maps_generator/README.md

The app version can be found in the "About" section in the settings menu of OMaps.

2. Build and install the generator_tool.

```sh

> ./tools/unix/build_omim.sh -r desktop

Reviewer: You don't need the desktop app for the generator.

./tools/unix/build_omim.sh -r generator_tool
./tools/unix/build_omim.sh -r world_roads_builder_tool
```

3. Change directory:

```sh
$ cd omim/tools/python/maps_generator
```

1. Install dependencies:

```sh
maps_generator$ pip3 install -r requirements_dev.txt
```

> 1 Make the ini configuration file:

Reviewer: Why 1. ?

Did you preview it? Was the markdown with numbered items rendered correctly?

Author: Whoops, fixing.

```sh
maps_generator$ cp var/etc/map_generator.ini.default var/etc/map_generator.ini
```

5. Edit the ini file:

```sh
maps_generator$ vim var/etc/map_generator.ini
```

> Here is a sample ini that will download an OSM PBF file for the Yukon Territories, Canada from [GEOFABRIK](https://www.geofabrik.de/). You can replace the *osm.pbf* and *osm.pbf.md5* with other areas instead. Note that an entire planet file currently takes 40+hrs on a server with 256GB of RAM. Unless you are have a machine this large, it is recommended to use a smaller extract.

Reviewer: ```suggestion
Here is a sample ini that will download an OSM PBF file for the Yukon Territories, Canada from [GEOFABRIK](https://www.geofabrik.de/). You can replace the *osm.pbf* and *osm.pbf.md5* with other areas instead. Note that an entire planet file currently takes 40+ hours on a server with 256GB of RAM. Unless you have a machine this large, it is recommended to use a smaller extract.
```

[Developer]
# The path where the generator_tool will be searched.
# Usually this is in the same parent directory of the organicmaps repo.
BUILD_PATH: ~/omim-build-release
# The path to the organicmaps repo
OMIM_PATH: ~/code/organicmaps

[Generator tool]

> # The path to the application data.

Reviewer: ```suggestion
# The path to the data folder in the repository.
```

# UK_POSTCODES_URL:
# US_POSTCODES_URL:

[Common]
# Auto detection.
THREADS_COUNT: 0

[Stats]
# Path to rules for calculating statistics by type
STATS_TYPES_CONFIG: ${Developer:OMIM_PATH}/tools/python/maps_generator/var/etc/stats_types_config.txt
```

### Notes

In each field where you need to specify a URL, you can specify the path to the file system using `file:///path/to/file`.

You can manually generate a subway layer file with the SUBWAY_URL parameter. See [instructions](https://github.com/organicmaps/organicmaps/tree/master/docs/SUBWAY_GENERATION.md).

## Basic Usage

Make sure you are in the `tools/python` directory when running the CLI.  Unless you have URLs for coastline files, skip the coastline.

```sh
cd tools/python
python -m maps_generator --countries="Canada_Yukon_North, Canada_Yukon_Whitehorse" --skip="Coastline"
```

## Help

```sh

> python$ python3.6 -m maps_generator -h

Reviewer: ```suggestion
python$ python -m maps_generator -h
```

  --order ORDER         Mwm generation order.
```

> If you are not from the OrganicMaps team, then you do not need the option --production when generating maps.

Reviewer: ```suggestion
If you are not from the Organic Maps team, then you do not need the option --production when generating maps.
```

PLANET_URL: https://download.geofabrik.de/asia/japan-latest.osm.pbf
PLANET_MD5_URL: https://download.geofabrik.de/asia/japan-latest.osm.pbf.md5
...

> To build an entire country with coastlines, you need to download the *latest_coasts.geom* and *latest_coasts.rawgeom* files and specify their path in the config.

Reviewer: It is only needed in the case of building some regions. If the whole planet is built, then these files will be generated/used directly.

Author: Are you saying that this part of the config can be left blank in a planet build? For this example, we are doing all of Japan. 

I have found that the generator fails if I include World and WorldCoasts, at least for the Yukon. So aren't the geom files always required?

Reviewer: You don't need to download these files if the whole planet is built. They are generated in the process of building the whole planet.

```

3. Run

```sh
python$ python -m maps_generator --countries="World, WorldCoasts, Japan_*"
```

### Rebuild stages

For example, you changed routing code in the project and want to regenerate maps.
You must have previous generation. You may regenerate from stage routing only for two mwms:

```sh

> python$ python3.6 -m maps_generator -c --from_stage="Routing" --countries="Japan_Kinki Region_Osaka_Osaka, Japan_Chugoku Region_Tottori"

Reviewer: ```suggestion
python$ python -m maps_generator -c --from_stage="Routing" --countries="Japan_Kinki Region_Osaka_Osaka, Japan_Chugoku Region_Tottori"
```

Please check if there are other 3.6 hardcoded versions somewhere.

3. Run

```sh

> python$ python -m maps_generator --countries="Russia_Moscow" --skip="Coastline"

Reviewer: ```suggestion
python$ python -m maps_generator --countries="Japan_Chugoku Region_Tottori" --skip="Coastline"
```
